src/Hazelcast.Net/Core/ClientVersion.cs

```diff
@@ -53,5 +57,33 @@ internal static string Version
                 return _clientVersion;
             }
         }
+
+        /// <summary>
+        /// Gets the "pure" version of this assembly.
```

Comment:
How does this change relate to this PR?

Reply:
Yes this relate to this PR because we send the client version to MC, and a .NET client version is `4.2.6+a3cde44` which is valid semver but MC does not like it and "dies" and we need to send the "pure" version ie `4.2.6`. See changes in `ClientMetricSource`.